### PR TITLE
Add Discord OAuth2 integration

### PR DESCRIPTION
**Pull request by amishshah**
Allows users to join the Hackathon Discord server using Discord OAuth2. The process adds them to the Hackathon server, no invite is needed.

Had to add some env variables:
```
# hs_discord_bot_api
DISCORD_HMAC_KEY="Random string, but must match the string in hs_discord_bot_api"
DISCORD_CLIENT_ID="692...412"
DISCORD_HS_API_BASE="http://localhost:8082"
```

`DISCORD_HMAC_KEY` is used to [provide security](https://auth0.com/docs/protocols/oauth2/oauth-state) in the `state` parameter of the OAuth2 call. It should be the same as the key used in `hs_discord_bot_api`.

`DISCORD_CLIENT_ID` stores the client ID of the Hackathon bot.

### CODE REVIEW

**Comment by amishshah** [comment]
Some screenshots so you can see the changes:
image
image
image

**Comment by amishshah** [comment]
Another thing to note is that the link in the navbar is unique to each user, if one user shared the link to another user, they wouldn't be able to authenticate.

File: src/views/partials/navbar.ejs [hunk 1]
```diff
@@ -62,6 +62,12 @@
           <p>Emergency Contacts</p>
         </a>
       </li>
+      <li class="nav-item">
```
Comment (seanjparker) [inline src/views/partials/navbar.ejs:65]: Can you add a check here so that we only show this navbar group if `locals.discordAuthURL` is defined?

**Comment by amishshah** [comment]
I changed it so that `/user/join_discord` is now a permalink to joining the Discord server. The hash generation is moved to this part as well, so that hashes aren't generated whenever the user loads any page. This link will redirect the user (provided they're logged in) to their personal auth link.

**Pull request merged**